doc/source/ray-contribute/development.rst

```diff
@@ -10,6 +10,68 @@ For a majority of Ray users, installing Ray via the latest wheels or pip package
 .. contents::
   :local:
 
+Clone the repository
+--------------------
+
+To build Ray locally you will need to have the Git repository, so first, fork it on GitHub.
+
+Then you can clone it to your machine:
+
+.. code-block:: shell
+
+    git clone https://github.com/[your username]/ray.git
+
+Next make sure you connect your repository to the upstream (main project) Ray repository. This will allow you to push your code to your repository when proposing changes (in pull requests) while also pulling updates from the main project:
+
+.. code-block:: shell
+    
+    # Enter into the cloned repository directory
+    cd ray
+    # Connect to the upstream repository
+    git remote add upstream https://github.com/ray-project/ray.git
+
+Every time you want to update your local version you can pull the changes from the main repository:
+
+.. code-block:: shell
+
+    # Checkout the local master branch
+    git checkout master
+    # Pull the latest changes from the main repository
+    git pull upstream master
+
+Prepare the Python environment
+------------------------------
+
+(Optional) To setup an isolated Anaconda environment, see :ref:`ray_anaconda`.
+
+In any case, you will probably want to have some type of Python virtual environment.
```

Comment:
Maybe we shouldn't make every sentence a new paragraph? I think this renders / reads not that great.

Reply:
Sure! I merged this. I think this also applies to other places so I merged other paragraphs too. 🤓